sentry/formatters.go

```diff
@@ -0,0 +1,67 @@
+package sentry
+
+import (
+	"strings"
+
+	"github.com/getsentry/sentry-go"
+	"golang.org/x/xerrors"
+)
+
+// headline returns a good Headline for this error.
+// Ideally, it returns a succinct summary that best conveys the error.
+// Most likely, that's something close to the root cause, but that may
+// be something boring like "context canceled".
+func headline(err error) string {
+	// Heuristic: return the error message from the second innermost error.
+	// This provides context on the error, since returned errors are often constants.
+	var prev error
+	for {
+		wrapper, ok := err.(xerrors.Wrapper)
+		if !ok {
+			break
+		}
+		prev = err
+		err = wrapper.Unwrap()
+	}
+	if prev != nil {
+		return prev.Error()
+	}
+	return err.Error()
+}
+
+
+// removeGlogPrefixFromMessage removes the glog date/level from the
+// raw byte string returned from glogEvent.Message
+func removeGlogPrefixFromMessage(msg []byte) string {
+	message := string(msg)
+	if square := strings.Index(message, "] "); square != -1 {
+		message = message[square+2:]
+	}
+
+	return message
+}
+
+// cleanupMessage cleans up a message displayed as the top-line
+// Sentry error by splitting at the first newline.
+func cleanupMessage(msg string) string {
+	return strings.Split(strings.TrimSpace(msg), "\n")[0]
+}
+
+// prependMessage prepends the given prefix to an existing message,
+// not prepending it if it already exists as a prefix
+func prependMessage(existing, prefix string) string {
+	prefix = strings.TrimPrefix(prefix, existing)
+	prefix = strings.TrimSpace(prefix)
+	prefix = strings.TrimPrefix(prefix, ":")
+	if len(prefix) > 0 {
+		return existing + "\n" + prefix
```

Comment:
Should it be: 

    return prefix + "\n" + existing
?

Reply:
I think I screwed up the variable names here. This is called like:
```
prependMessage(hl, s.Message)
```

where hl is the "headline" of the error, i.e. an abbreviated form of s.Message. So in that case "existing" is the headline, while "prefix" is the full message, so the names should actually be swapped (but the variable order is correct).

Reply:
Force-pushed an update that clarifies the variable names + docstring